File: src/main/java/de/steklopod/sec03/Lec05FluxGenerate.java
package de.steklopod.sec03;

import de.steklopod.utils.Util;
import reactor.core.publisher.Flux;

public class Lec05FluxGenerate {

    public static void main(String[] args) {

        Flux.generate(synchronousSink -> {
            System.out.println("emitting");
            synchronousSink.next(Util.faker().country().name()); // 1
            //synchronousSink.error(new RuntimeException("oops"));
        })
        .take(2)
        .subscribe(Util.subscriber());


    }

}

File: src/main/kotlin/de/steklopod/sec03/Lec05FluxGenerate.kt
package de.steklopod.sec03

import de.steklopod.utils.Util.faker
import de.steklopod.utils.Util.subscriber
import reactor.core.publisher.Flux
import reactor.core.publisher.SynchronousSink

object Lec05FluxGenerate {
    @JvmStatic
    fun main(args: Array<String>) {
        Flux.generate { synchronousSink: SynchronousSink<Any> ->
            println("emitting")
            synchronousSink.next(faker().country().name()) // 1
        }
            .take(2)
            .subscribe(subscriber())
    }
}
